### PR TITLE
Refactor InsightsService toward modularity and fix categories bug

### PR DESCRIPTION
**Pull request by nickhurlburt**
Primary reviewer: @GPaoloni 
This PR may be easier to review by commit.  Here are the highlights:
* Adds back use of the `customer` object in Insights
* Fixes an issue with the categories not being converted to a string correctly
* Adds a test for contactless tasks
* Converts InsightsService to TS (the individual commit shows `git mv` but the full diff does not)
* Refactors InsightsService to apply a series of updates

This is an intermediate PR toward accomplishing [CHI-383](https://bugs.benetech.org/browse/CHI-383).  The refactoring and added testing will help to add additional helpline-specific insights updates.

### CODE REVIEW

**Review by GPaoloni — APPROVED** [review]
Looks awesome! Loved reviewing this commit by commit and seen how this converted into functional, idiomatic and TS.

File: plugin-hrm-form/src/services/InsightsService.js [hunk 1]
```diff
@@ -11,6 +11,18 @@ function getSubcategories(form) {
   return categories.splice(0, 3).join(';');
 }
 
+function buildCustomersObject(taskAttributes, form) {
+  const callType = form.callType.value;
```
[the remark this answers is not in the document]
Comment (GPaoloni) [reply plugin-hrm-form/src/services/InsightsService.js]: Solved in a following commit.

File: plugin-hrm-form/src/services/InsightsService.js [hunk 2]
```diff
@@ -11,6 +11,18 @@ function getSubcategories(form) {
...
+  const hasCustomerData = !isNonDataCallType(callType);
+
+  if (!hasCustomerData) return {};
+
+  const { childInformation } = form;
+  return {
+    gender: childInformation.gender.value,
```
[the remark this answers is not in the document]
Comment (GPaoloni) [reply plugin-hrm-form/src/services/InsightsService.js]: Solved in a following commit.

File: plugin-hrm-form/src/services/InsightsService.ts [hunk 3]
```diff
@@ -39,56 +46,67 @@ function buildConversationsObject(taskAttributes: TaskAttributes, contactForm: T
...
-  const previousAttributes = twilioTask.attributes;
-  const mergedAttributes = mergeAttributes(previousAttributes, { conversations, customers });
-  const finalAttributes = previousAttributes.isContactlessTask
-    ? overrideAttributes(mergedAttributes, contactForm)
-    : mergedAttributes;
+  const conversations : InsightsAttributes = buildConversationsObject(twilioTask.attributes, contactForm);
+  const customers : InsightsAttributes = buildCustomersObject(twilioTask.attributes, contactForm);
+  const contactlessAttributes : InsightsAttributes = overrideAttributes(twilioTask.attributes, contactForm);
+  const previousAttributes : TaskAttributes = twilioTask.attributes;
+  const finalAttributes : TaskAttributes = mergeAttributes(mergeAttributes(mergeAttributes(previousAttributes, conversations), customers), contactlessAttributes);
```
[the remark this answers is not in the document]
Comment (GPaoloni) [reply plugin-hrm-form/src/services/InsightsService.ts]: Solved in a following commit.

File: plugin-hrm-form/src/services/InsightsService.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nickhurlburt) [reply plugin-hrm-form/src/services/InsightsService.js]: Yes, that got fixed later.

File: plugin-hrm-form/src/services/InsightsService.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nickhurlburt) [reply plugin-hrm-form/src/services/InsightsService.js]: Yes, that got fixed later.

File: plugin-hrm-form/src/services/InsightsService.ts [hunk 4]
```diff
@@ -2,33 +2,39 @@
...
 import { mapChannelForInsights } from '../utils/mappers';
 import { getDateTime } from '../utils/helpers';
+import { TaskEntry } from 'states/contacts/reducer';
+import { ITask } from '@twilio/flex-ui';
 
-function getSubcategories(form) {
-  if (!form || !form.categories) return [];
+// How do we do this?  Twilio docs just define it as an Object
+// See https://assets.flex.twilio.com/releases/flex-ui/1.18.0/docs/ITask.html
+type TaskAttributes = any;
```
[the remark this answers is not in the document]
Comment (nickhurlburt) [reply plugin-hrm-form/src/services/InsightsService.ts:14]: Sigh.  Maybe I can go with `object`.

File: plugin-hrm-form/src/services/InsightsService.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (nickhurlburt) [reply plugin-hrm-form/src/services/InsightsService.ts]: Yes, I hacked together that initial commit and then used `reduce` on a later one.

**Comment by nickhurlburt** [comment]
Deployed to staging, after adding quick fix #296

**Pull request merged**